src/network/peer.zig

```diff
@@ -90,6 +90,35 @@ pub const Peer = struct {
 
     /// Send version message to peer
     fn sendVersionMessage(self: *Peer) !void {
+        if (self.address.any.family == std.posix.AF.INET) {
+            // Convert IPv4 to IPv6-mapped IPv4 address
+            const ipv4_address = self.address.in;
+
+            var ipv6_mapped: [16]u8 = [_]u8{0} ** 16;
+
+            // Set bytes 10 and 11 to 0xff
+            ipv6_mapped[10] = 0xff;
+            ipv6_mapped[11] = 0xff;
+
+            // Copy the IPv4 address into the last 4 bytes of the IPv6-mapped address
+            const ipv4_bytes = std.mem.asBytes(&ipv4_address.sa.addr);
+            ipv6_mapped[12] = ipv4_bytes[0];
+            ipv6_mapped[13] = ipv4_bytes[1];
+            ipv6_mapped[14] = ipv4_bytes[2];
+            ipv6_mapped[15] = ipv4_bytes[3];
+
+            self.address = std.net.Address{
+                .in6 = .{
+                    .sa = std.posix.sockaddr.in6{
+                        .addr = ipv6_mapped,
+                        .port = ipv4_address.getPort(),
+                        .flowinfo = 0,
+                        .scope_id = 0,
+                    },
+                },
+            };
+        }
```

Comment:
this will be needed for other messages.
Move it as a standalone method under message/utils.zig